### PR TITLE
Ensure stage are reloaded on Refresh and Reload actions (USDU-58, #155)

### LINKED ISSUE #155: Inability to refresh layers in Editor Mode (FTV-657)
Edits to USD layers outside of Unity are not updated with Refresh in Editor Mode.

**Steps to reproduce:**

1. Import a cube animating along an axis, either as gameObject or Prefab (refresh does not work on either)
2. Add USD Playable Track to Timeline, reference the USD Asset, and add a track. Scrub timeline to play back the animated cube.
3. Make a change to the cube animation and save the USD layer. (Verify change has been saved by refreshing the layer in usdview).
4. Click Refresh USD on prefab or gameobject, depending on how it was imported in Step 1.
5. Scrub timeline in Editor and note animation has not changed from initial import. Enter Runtime mode and note the updated animation is loaded correctly.

**Desired behavior:** in Step 5 above, refreshing the USD Asset should reload the layer in Editor mode.

Workarounds: none known.
Unity version: tested in 2019.3.b07 and 2019.3.0f3
Unity USD version: 1.0.2-preview.1

### PR DESCRIPTION
**Pull request by lucillecaillaud**
## Purpose of this PR

**Ticket/Jira #:** [USDU-58](https://jira.unity3d.com/browse/USDU-58)

UsdStage.Open won't open again a layer that is already open (opened layers are cached). So if the layer has changed on disk, it needs either to be closed and re-open or simply reloaded to get the new changes.

## Testing

**Functional Testing status:**
Added 3 unit tests to ensure the file is reloaded when refreshing or reloading via the UsdAsset.
* Test that nothing happen if the file didn't change on disk.
* Test that the new attribute's values can be retrieved when the file change on disk and user refresh it.
* Test that the new attribute's values can be retrieved when the file change on disk and user reload it.

**Performance Testing status:**
N/A

## Overall Product Risks

**Complexity:** Minimal

**Halo Effect:** Minimal

## Additional information

**Note to reviewers:**

**Reminder:**

- [x] Add entry in CHANGELOG.md _(if applicable)_

### CODE REVIEW

**Review by judubu — CHANGES_REQUESTED** [review]
Good catch. 
Can you write a quick test where you open file with a simple mesh, modify the displayColor or the transform, overwrite the file and reload?

File: package/com.unity.formats.usd/Runtime/Scripts/Behaviors/UsdAsset.cs [hunk 1]
```diff
@@ -394,6 +394,7 @@ public string usdFullPath
           stage = pxr.UsdStage.Open(usdFullPath, pxr.UsdStage.InitialLoadSet.LoadAll);
         }
 
+        stage.Reload();  // Ensure the stage is reloaded it case it was already opened and cached.
```
Comment (vkovec) [inline package/com.unity.formats.usd/Runtime/Scripts/Behaviors/UsdAsset.cs]: small typo: `reloaded it` should be `reloaded in`

**Review by vkovec — APPROVED** [review]

**Review by jcowles — APPROVED** [review]
Modulo the typo that was called out, looks good to me.

Also, +1 to Julien's proposal to make a test.

File: package/com.unity.formats.usd/Runtime/Scripts/Behaviors/UsdAsset.cs (hunk 1 above)
Comment (lucillecaillaud) [reply package/com.unity.formats.usd/Runtime/Scripts/Behaviors/UsdAsset.cs]: Fixed in https://github.com/Unity-Technologies/usd-unity-sdk/pull/216/commits/6aa01e5a0a82d457c391f5056b885b1fbf7f833a

**Comment by lucillecaillaud** [comment]
Added Unit tests in https://github.com/Unity-Technologies/usd-unity-sdk/pull/216/commits/6aa01e5a0a82d457c391f5056b885b1fbf7f833a

I wasn't able to use moq to simply Mock UsdStage.Reload and ensure the function is called when we need to. So I followed Julien's proposal of doing a swap of files on disk to simulate a change in the file from outside of the current context.

File: package/com.unity.formats.usd/Tests/Editor/Unity.Formats.USD.Tests.Editor.asmdef [hunk 2]
```diff
@@ -2,16 +2,25 @@
...
         "Editor"
     ],
+    "excludePlatforms": [],
     "allowUnsafeCode": true,
+    "overrideReferences": true,
     "precompiledReferences": [
-        "USD.NET.dll"
-    ]
+        "USD.NET.dll",
+        "USD.NET.Unity.dll",
```
Comment (lucillecaillaud) [inline package/com.unity.formats.usd/Tests/Editor/Unity.Formats.USD.Tests.Editor.asmdef]: I actually only add this dll. Not sure why there is so much that changed. I guess assemblies serialization changed.

**Review by cguer — APPROVED** [review]
QA approved 👍!

Tested on:
- 2019.4.13f1 (Windows only)

File: package/com.unity.formats.usd/Tests/Editor/Unity.Formats.USD.Tests.Editor.asmdef (hunk 2 above)
Comment (judubu) [reply package/com.unity.formats.usd/Tests/Editor/Unity.Formats.USD.Tests.Editor.asmdef]: No worries, I reverted the unwanted changes.

**Review by judubu — APPROVED** [review]
All good, thanks for the test.
I made the small change to the tests to let the editor tick and register the file changes on disk. Without that it looks like the os was not registering the file moves.

**Review by judubu — APPROVED** [review]
Thanks @thomas-tu for the extra pass on the tests :+1:

**Review by thomas-tu — APPROVED** [review]

**Pull request merged**